contracts/TransferLimitModule.sol

```diff
@@ -0,0 +1,417 @@
+    /// @param amount Amount of tokens (or Wei) that should be transfered
+    /// @param _nonce Nonce used for this Safe transaction.
+    /// @param safeTxGas Gas that should be used for the Safe transaction.
+    /// @param dataGas Gas costs for data used to trigger the safe transaction and to pay the payment transfer
+    /// @param gasPrice Gas price that should be used for the payment calculation.
+    /// @param gasToken Token address (or 0 if ETH) that is used for the payment.
+    /// @param refundReceiver Address of receiver of gas payment (or 0 if tx.origin).
+    /// @param _nonce Transaction nonce.
+    /// @return Transaction hash.
+    function getTransactionHash(
+        address token,
+        address to,
+        uint256 amount,
+        uint256 safeTxGas,
+        uint256 dataGas,
+        uint256 gasPrice,
+        address gasToken,
+        address refundReceiver,
+        uint256 _nonce
+    )
+        public
+        view
+        returns (bytes32)
+    {
+        return keccak256(
+            abi.encodePacked(byte(0x19), byte(0), this, token, to, amount, safeTxGas, dataGas, gasPrice, gasToken, refundReceiver, _nonce)
+        );
+    }
+
+    /// @dev Returns start of the current time period.
+    /// @return Unix timestamp.
+    function currentStartTime()
+        public
+        view
+        returns (uint)
+    {
+        return getNow() - (getNow() % timePeriod);
+    }
+
+    function handleTransferLimits(address token, uint256 amount)
+        internal
+        returns (bool)
+    {
+        TransferLimit storage transferLimit = transferLimits[token];
+        // If time period is over, reset expenditure.
+        if (isPeriodOver()) {
+            transferLimit.spent = 0;
+            totalWeiSpent = 0;
+            totalDaiSpent = 0;
+        }
+
+        // Transfer + previous expenditure shouldn't exceed limit specified for token.
+        if (SafeMath.add(transferLimit.spent, amount) > transferLimit.transferLimit) {
+            return false;
+        }
+
+        // If a global cap is set, transfer amount + value of all
+        // previous expenditures (for all tokens) shouldn't exceed global limit.
+        if (!isUnderGlobalCap(token, amount)) {
+            return false;
+        }
+
+        transferLimits[token].spent = SafeMath.add(transferLimits[token].spent, amount);
+
+        return true;
+    }
+
+    function isUnderGlobalCap(address token, uint256 amount)
+        internal
+        returns (bool)
+    {
+        if (globalWeiCap == 0 && globalDaiCap == 0) {
+            return true;
+        }
+
+        // Calculate value in ether.
+        uint256 ethNum;
+        uint256 ethDen;
+        (ethNum, ethDen) = getEthAmount(token, amount);
+
+        // Convert ether to wei
+        uint256 weiAmount = SafeMath.div(SafeMath.mul(ethNum, 10**18), ethDen);
+        if (globalWeiCap > 0 && SafeMath.add(totalWeiSpent, weiAmount) > globalWeiCap) {
```

Comment:
we could store `SafeMath.add(totalWeiSpent, weiAmount)` to avoid calculating it twice ... but not super important (might save a couple gas ... not sure)